Barkane/Assets/Scripts/VFX/JointRenderer.cs

```diff
@@ -0,0 +1,517 @@
+                {
+                }
+                // the other pairing
+                else if (CoordUtils.RoundEquals(a1.transform.up, -b2.transform.up))
+                {
+                    this.b1 = b2;
+                    this.b2 = b1;
+                }
+                // invalid
+                else throw new UnityException("Cannot pair a1 with anything! (Overlapping Case)");
+
+                foldState = FoldState.Overlap;
+                break;
+            
+            // for coplanar case, just compare if the normals match up
+            case 1:
+                // pair a1 with b1, a2 with b2
+                if (CoordUtils.RoundEquals(a1.transform.up, b1.transform.up))
+                {
+
+                }
+                // the other pairing
+                else if (CoordUtils.RoundEquals(a1.transform.up, b2.transform.up))
+                {
+                    this.b1 = b2;
+                    this.b2 = b1;
+                }
+                // invalid
+                else throw new UnityException("Can't pair a1 with anything! (Coplanar Case)");
+
+                foldState = FoldState.Coplanar;
+                break;
+            case 2:
+                // for orthogonal case, compare if the "towards" direction is in the same way as the normal (for both sides)
+                // this compares whether two sides are facing "inwards" or "outwards"
+                var a2b = CoordUtils.AsV(CoordUtils.FromTo(a1, b1));
+                var b2a = -a2b;
+                
+                // pair a1 with b1, a2 with b2
+                if (Mathf.Sign(Vector3.Dot(a1.transform.up, a2b)) == Mathf.Sign(Vector3.Dot(b1.transform.up, b2a)))
+                {
+
+                }
+                // the other pairing
+                else if (Mathf.Sign(Vector3.Dot(a1.transform.up, a2b)) == Mathf.Sign(Vector3.Dot(b1.transform.up, a2b)))
+                {
+                    this.b1 = b2;
+                    this.b2 = b1;
+                }
+                // invalid
+                else throw new UnityException("Can't pair a1 with anything! (Orthogonal Case)");
+
+                foldState = FoldState.Orthogonal;
+                break;
+            
+            // somehow there are more than 3 axis different in a 3D space..?
+            default:
+                // this includes the possible 3 case when non of the coordinates are equal (the tiles aren't even adjacent!)
+                throw new UnityException($"Joint { transform.parent } contains squares that aren't adjacent!");
+        }
+    }
+
+    private void UpdateColors()
+    {
+        colorA1 = a1.EdgeTintedColor(settings.tintCorrection);
+        colorA2 = a2.EdgeTintedColor(settings.tintCorrection);
+        colorB1 = b1.EdgeTintedColor(settings.tintCorrection);
+        colorB2 = b2.EdgeTintedColor(settings.tintCorrection);
+    }
+
+    private static readonly float squareSize = 2;
```

Comment:
should be at top of class

Reply:
refactored this to SquarerRenderSettings